src/prefect/client/client.py

```diff
@@ -315,6 +319,12 @@ def _request(
 
         return response
 
+    def attach_headers(self, headers: dict) -> None:
+        """
+        Set headers to be attached to this Client
+        """
+        self._attached_headers = headers  # type: ignore
```

Comment:
In case we want to use this more in the future we might want to store all attached headers as an attribute and update them with each call to this method (so that multiple calls don't wipe old headers but rather add to them / update them)